src/workers/generateAsync.worker.js

```diff
@@ -43,6 +43,12 @@ global.onmessage = function ( { data } ) {
 
 		}
 
+		if ( options.groups ) {
+
+			geometry.groups = options.groups;
+
+		}
```

Comment:
yes, more automatic might be nice.
didt really think about it since I use premerged meshed and have group info somewhere else. Only reason why I add it is to give it to mesh-bvh so I can later generate a "selection" mesh 😁

Reply:
Great thanks for the change :)